napari/utils/events/event_utils.py

```diff
@@ -0,0 +1,18 @@
+def disconnect_events(source, listener):
+    """Disconnect all events between a source and a listener.
+
+    Parameters
+    ----------
+    source : Object
+        Any object with an event emitter.
+    listener: Object
+        Any object that has been connected too.
+    """
+    for em in source.events.emitters.values():
+        for callback in em.callbacks:
+            # Callback is a tuple of a weak reference and method name
+            if (
+                isinstance(callback, tuple)
```

Comment:
It could be callable directly, which means it can't have been attached to the listener.

I've added this comment

```
            *callback* may be either a callable object or a tuple
            (object, attr_name) where object.attr_name will point to a
            callable object. Note that only a weak reference to ``object``
            will be kept. If *callback* is a callable object then it is
            not attached to the listener and does not need to be
            disconnected
```